src/unix.rs

```diff
@@ -368,6 +392,83 @@ impl Client {
     }
 }
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod linux {
+    use super::*;
+
+    use libc::{iovec, off_t, ssize_t, syscall, SYS_preadv2};
+
+    // TODO: Replace this with libc::RWF_NOWAIT once they have it for musl
+    //  targets
+    const RWF_NOWAIT: c_int = 0x00000008;
+
+    fn cvt_ssize(t: ssize_t) -> io::Result<ssize_t> {
+        if t == -1 {
+            Err(io::Error::last_os_error())
+        } else {
+            Ok(t)
+        }
+    }
+
+    fn preadv2(fd: c_int, iov: &iovec) -> ssize_t {
+        let iovcnt: c_int = 1;
+
```

Comment:
I meant something more like,
```suggestion
        let offset = -1; // "use and update current offset" as in readv, not preadv

```

bit of a magic number otherwise.

Reply:
Thanks, I've added the `offset` variable with the right typing.